unstructured_ingest/v2/processes/connectors/redisdb.py

```diff
@@ -0,0 +1,209 @@
+    )
+    host: Optional[str] = Field(
+        default=None, description="hostname or IP address of a single redis instance to connect to"
+    )
+    database: Optional[int] = Field(default=0, description="database index to connect to")
+    port: Optional[int] = Field(default=6379)
+    username: Optional[str] = Field(default=None)
+    ssl: Optional[bool] = Field(default=True)
+    connector_type: str = Field(default=CONNECTOR_TYPE, init=False)
+
+
+class RedisUploadStagerConfig(UploadStagerConfig):
+    pass
+
+
+@dataclass
+class RedisUploadStager(UploadStager):
+    upload_stager_config: RedisUploadStagerConfig = field(
+        default_factory=lambda: RedisUploadStagerConfig()
+    )
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        shutil.copy(elements_filepath, output_path)
+        return output_path
+
+
+class RedisUploaderConfig(UploaderConfig):
+    batch_size: int = Field(default=100, description="Number of records per batch")
+
+
+@dataclass
+class RedisUploader(Uploader):
+    upload_config: RedisUploaderConfig
+    connection_config: RedisConnectionConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    def is_async(self) -> bool:
+        return True
+
+    def precheck(self) -> None:
+        if not self.connection_config.access_config.uri and not self.connection_config.host:
+            raise ValueError("Please pass a hostname either directly or through uri")
+
+        try:
+            client = self.create_client()
+            client.ping()
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise DestinationConnectionError(f"failed to validate connection: {e}")
+
+    @requires_dependencies(["redis"], extras="redis")
+    def create_client(self, async_flag=Field(default=True)) -> "Redis":
+        if async_flag:
+            from redis.asyncio import Redis, from_url
+        else:
+            from redis import Redis, from_url
+
+        access_config = self.connection_config.access_config.get_secret_value()
+
+        if access_config.uri:
+            return from_url(access_config.uri)
+        elif access_config.password:
+            return Redis(
+                host=self.connection_config.host,
+                port=self.connection_config.port,
+                db=self.connection_config.database,
+                password=access_config.password,
+                ssl=self.connection_config.ssl,
+            )
+        else:
+            return Redis(
+                host=self.connection_config.host,
+                port=self.connection_config.port,
+                db=self.connection_config.database,
+                ssl=self.connection_config.ssl,
+            )
```

Comment:
When repeating multiple parameters like this you can alternatively do something along the lines of:
```py
options = {"host": self.connection_config.host, "port": self.connection_config.port, ...}
if access_config.password:
    options["password"] = access_config.password
return Redis(**options)
```
it's not necessary but something to consider.